src/WebJobs.Script/StorageProvider/AzureStorageProvider.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Azure.Storage.Blobs;
+using Microsoft.Azure.WebJobs.StorageProvider.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    /// <summary>
+    /// This serves as a placeholder for a concrete implementation of a storage abstraction.
+    /// This StorageProvider provides a wrapper to align all uses of storage by the Functions Host
+    /// </summary>
+    internal class AzureStorageProvider : IAzureStorageProvider
+    {
+        private readonly BlobServiceClientProvider _blobServiceClientProvider;
+        private readonly IConfiguration _configuration;
+        private IOptionsMonitor<JobHostInternalStorageOptions> _storageOptions;
+
+        public AzureStorageProvider(IConfiguration configuration, BlobServiceClientProvider blobServiceClientProvider, IOptionsMonitor<JobHostInternalStorageOptions> options)
+        {
+            _blobServiceClientProvider = blobServiceClientProvider;
+            _configuration = configuration;
+            _storageOptions = options;
+        }
+
+        /// <summary>
+        /// Try create BlobServiceClient
+        /// SDK parse connection string method can throw exceptions: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/Shared/StorageConnectionString.cs#L238
+        /// </summary>
+        /// <param name="client">client to instantiate</param>
+        /// <param name="connectionString">connection string to use</param>
+        /// <returns>successful client creation</returns>
+        public virtual bool TryGetBlobServiceClientFromConnectionString(out BlobServiceClient client, string connectionString = null)
+        {
+            try
+            {
+                var connectionStringToUse = connectionString ?? _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
+                _blobServiceClientProvider.TryGetFromConnectionString(connectionString, out client);
+                return true;
+            }
+            catch
```

Comment:
This code confuses me a little - you're saying TryGetFromConnectionString can throw exceptions because parsing the connection string fails? I would expect it to return false - otherwise it doesn't behave how we'd typically expect for a "TryFoo" method. Any reason you're not checking the return value? Can the underlying implementation use StorageConnectionString.TryParse?

Reply:
Ah. I overlooked when I went back and forth deciding where the actual try/catch logic should live. I've updated to just use the logic in StorageClientProvider in webjobs. This makes these methods in AzureStorageProvider a lot simpler.